### PR TITLE
Add streaming interface

### PR DESCRIPTION
**Pull request by floatdrop**
I'm pretty sure, that there is better way to extend `got`. Any ideas?

### CODE REVIEW

File: test.js [hunk 1]
```diff
@@ -18,3 +19,8 @@ test('endpoint option', t => {
 	// TODO: use `t.throws` when it supports promises
 	return fn('users/sindresorhus', {endpoint: 'fail'}).catch(err => t.ok(err));
 });
+
+test('stream interface', async t => {
+	t.regexTest(/"login":"sindresorhus"/, await getStream(fn.stream('users/sindresorhus')));
```
Comment (sindresorhus) [inline test.js]: Just JSON.parse and check the property

**Comment by sindresorhus** [comment]
The stream property should probably be documented. It's not very clear it can be used.

**Comment by sindresorhus** [comment]
> I'm pretty sure, that there is better way to extend got. Any ideas?

No that I can think of right now.

// @kevva @SamVerschueren

**Comment by SamVerschueren** [comment]
Couldn't think of an easier way (for now).

**Comment by kevva** [comment]
Nope, this looks good to me too.

**Pull request merged**